### ISSUE TITLE
Investigate performance issues in V1

### ISSUE THREAD

**Issue by matof-who** [HeRAMS-WHO/herams-backlog#149]
**Description of the issue**

Performance on production site is highly degraded when there are many users connected

Today, with around 100 users connected login took more 30 seconds and many users called to complain about "gateway timeout" error message. The slowness affects all pages in the platform.

When the number of users connected goes lower, the performance of the site is better

The problem seems to be related about how the permissions are calculated for the connected users

**Goal**
The goal of this issue to analyse in detail what is causing the performance issue, and to analyse different solutions (if there are any)

- [ ] Replicate in local environment the volume of data in production
- [ ] Analyse the root cause of the slow performance
- [ ] Document possible solutions with pros/cons, regression risk and effort needed

**Comment by matof-who** [comment]
Cristian and myself have been analyzing the root cause for the performance issues in PROD. After looking at the database usage, we have detected two queries that were very slow and caused the timeouts

- The homepage query. In particular, calculating the number of contributors for each project
- The workspace list query. In particular, calculating the last update date of each workspace

We have added some indexes to the prime2_permissions table and this seems to have a very positive effect in the performance.

- one on source colument
- one on (target, target_id) columns

We will continue following up tomorrow to see the performance issues reappear.
Another possible “quick-fix” we’ve discussed with Raphael would be to temporary hide the last update date form the workspace list. We will proceed with that change if the performance issue reappear.
